### DISCUSSION TITLE
How to inherit Dockerfile environment variables into a VScode terminal?

### DISCUSSION THREAD

**Discussion by amoosebitmymom** [coder/code-server#6838] [Q&A]
I am running inside an air-gapped environment. 
I have a Dockerfile where I install code-server, and run an ssh daemon to connect to. 
Inside the Dockerfile, I configure many important variables. For instance, I am running my own artifactory server, so I set PIP environment variables so that pip could work (I know that I can put pip values in a config file, but some stuff I can't).

When I connect to the code-server from vscode (using Remote SSH) and open a new terminal, none of the environment variables I set. I understand why this is the behavior.
Is there a way to configure in coder to inherit these environment variables?
I know I can write a terminal profile with environment variables in it. But then every variable I define in the Dockerfile, I have to define separately in the settings.json. These separate sources of truths could cause configuration drift overtime. I'd rather not pursue that solution unless absolutely necessary 

Any help would be appreciated :)

**Comment by code-asher** [comment]
Since you mention Remote SSH, you are probably talking about VS Code;  code-server is a browser-only wrapper around VS Code and does not support the Remote SSH extension, so you might want to ask up at https://github.com/microsoft/vscode.

That said, I think adding the variables to the profile is the best option.  Maybe you can do something with `docker --env-file`, and mount that same file to read in the profile.

**Reply by amoosebitmymom** [reply]
Thanks for the reply, however after further research I realized my issue isn't related at all to this extension. 

When you SSH into a container, sshd opens a new shell which is purposely disconnected from the other shells. 
In order to propagate my environment variables, I need to either configure them in config files (instead of env vars), or put them in a file run at startup (such as /etc/profile).

On a completely different note, what do you mean by Remote SSH is not supported? Because I do use that extension, and when I use the code-server CLI (for instance to install extensions), or when I change the settings.json, it works.
Is that not intended behavior?

**Reply by code-asher** [reply]
>  I need to either configure them in config files (instead of env vars), or put them in a file run at startup (such as /etc/profile).

Yup makes sense!

> what do you mean by Remote SSH is not supported

### Some background first

Microsoft has a desktop version and a web/browser version of VS Code, the source code of both which are available at https://github.com/microsoft/vscode.  The web version can be packaged standalone or with a remote server component.  That remote server component is also separately packaged to be used for the Remote SSH extension.

I believe Microsoft calls just the server component `code-server`, while we have named the web plus server package `code-server` (which is what we build in this repo after applying some patches), although to be fair I think we were the first to use the name.

### And then for the actual answer

If you try to install the Remote SSH extension directly in our code-server (via the browser), you get the error `This extension is disabled because it is defined to run only in code-server for the Desktop`.

Now, if you install our code-server in a remote machine, and then use the Remote SSH extension from desktop to that remote, our code-server is not used; the server component packaged with the web version is not reusable for Remote SSH connections.

The Remote SSH extension instead downloads the server component separately from Microsoft, runs that, then connects to it.  So in this scenario you end up with two slightly dissimilar instances of `code-server` running on the remote, one for the browser, and one for the remote extension.
